### PR TITLE
Legacy Settings & Keymapping Converter

### PR DESCRIPTION
**Pull request by IsaacMarovitz**

### CODE REVIEW

File: PlayCover/Utils/LegacySettings.swift [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+    static func convertLegacyKeymapFile(_ from: URL) -> Keymap? {
+        var dictionary: [String: Any]
+
+        do {
+            if let settings = try [String: Any].read(from) {
+                if !settings.isEmpty {
+                    dictionary = settings
+                    let legacyKeymaps = dictionary["pc.layout"] as? [String: Any] ?? [:]
+                    var keymap = Keymap(bundleIdentifier: "")
```
Comment (TheMoonThatRises) [inline PlayCover/Utils/LegacySettings.swift]: You can get the bundleIdentifier from
```suggest
from.deletingPathExtension().lastPathComponent
```

File: PlayCover/Utils/LegacySettings.swift [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        }
+    }
+
+    static func convertLegacySettingsDict(_ from: [String: Any]) -> AppSettingsData? {
+        var dictionary: [String: Any]
+
+        if !from.isEmpty {
+            dictionary = from
+
+            var settingsData = AppSettingsData()
```
Comment (TheMoonThatRises) [inline PlayCover/Utils/LegacySettings.swift:65]: The legacy keymap settings also has:
```suggest
pc.bypass, pc.enableWindowAutoSize, pc.gameWindowSizeHeight, pc.gameWindowSizeWidth, pc.ipadModel
```
which are stored in AppSettingsData

**Comment by ZhichGaming** [comment]
This error should have a more detailed description such as: Legacy keymapping file names must be the same as the application bundle ID to be imported. Other than that, I did not run into any major issues. 

image

File: PlayCover/Utils/LegacySettings.swift (hunk 2 above)
Comment (JoseMoreville) [reply PlayCover/Utils/LegacySettings.swift:65]: Those are not keymapping, those are settings that can be changed in menu of they're still available.

Comment (Depal1) [reply PlayCover/Utils/LegacySettings.swift:65]: This PR is aiming to preserve settings too.

Comment (IsaacMarovitz) [reply PlayCover/Utils/LegacySettings.swift:65]: Here's my rational for excluding those parameters.

`pc.bypass`: Jailbreak Bypass did not work as intended in the majority of legacy PlayCover
`pc.enableWindowAutoSize`, `pc.gameWindowSizeHeight`, `pc.gameWindowSizeWidth`: The way that resolution behaves in 2.0 has been almost entirely refactored. Setting new settings based on this data might cause unintended behaviour.
`pc.ipadModel`: M1 Model Identifier <2.0 was broken, now that has been fixed, the default has been changed to M1

**Review by JoseMoreville — APPROVED** [review]

**Pull request merged**